superset/assets/javascripts/dashboard/components/GridCell.jsx

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable react/no-danger */
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import SliceHeader from './SliceHeader';
+import ChartContainer from '../../chart/ChartContainer';
+
+import '../../../stylesheets/dashboard.css';
+
+const propTypes = {
+  timeout: PropTypes.number,
+  datasource: PropTypes.object,
+  isLoading: PropTypes.bool,
+  isExpanded: PropTypes.bool,
+  widgetHeight: PropTypes.number,
+  widgetWidth: PropTypes.number,
+  fetchSlice: PropTypes.func,
+  removeSlice: PropTypes.func.isRequired,
+  updateSliceName: PropTypes.func,
+  toggleExpandSlice: PropTypes.func,
+  slice: PropTypes.object,
+  chartKey: PropTypes.string,
+  formData: PropTypes.object,
+  filters: PropTypes.object,
+  addFilter: PropTypes.func,
+  clearFilter: PropTypes.func,
+  removeFilter: PropTypes.func,
+  exploreChartUrl: PropTypes.string,
+  exportCSVUrl: PropTypes.string,
+};
+
+class GridCell extends React.PureComponent {
+  getDescriptionId(slice) {
+    return 'description_' + slice.slice_id;
+  }
+
+  getHeaderId(slice) {
+    return 'header_' + slice.slice_id;
+  }
+
+  width() {
+    return this.props.widgetWidth - 10;
+  }
+
+  height(slice) {
+    const widgetHeight = this.props.widgetHeight;
+    const headerId = this.getHeaderId(slice);
+    const descriptionId = this.getDescriptionId(slice);
+    const headerHeight = this.refs[headerId] ? this.refs[headerId].offsetHeight : 30;
+    let descriptionHeight = 0;
+    if (this.props.isExpanded && this.refs[descriptionId]) {
+      descriptionHeight = this.refs[descriptionId].offsetHeight + 10;
+    }
+    return widgetHeight - headerHeight - descriptionHeight;
+  }
+
+  render() {
+    const slice = this.props.slice;
+    return (
+      <div
+        className={this.props.isLoading ? 'slice-cell-highlight' : 'slice-cell'}
+        id={`${slice.slice_id}-cell`}
+      >
+        <div ref={this.getHeaderId(slice)}>
+          <SliceHeader
+            slice={slice}
+            exploreChartUrl={this.props.exploreChartUrl}
+            exportCSVUrl={this.props.exportCSVUrl}
+            isExpanded={this.props.isExpanded}
+            removeSlice={this.props.removeSlice}
+            updateSliceName={this.props.updateSliceName}
+            toggleExpandSlice={this.props.toggleExpandSlice}
+            forceRefresh={() => this.props.fetchSlice(true)}
+          />
+        </div>
+        <div
+          className="slice_description bs-callout bs-callout-default"
+          style={this.props.isExpanded ? {} : { display: 'none' }}
+          ref={this.getDescriptionId(slice)}
+          dangerouslySetInnerHTML={{ __html: slice.description_markeddown }}
+        />
+        <div className="row chart-container">
+          <input type="hidden" value="false" />
+          <ChartContainer
+            containerId={`slice-container-${slice.slice_id}`}
+            chartKey={this.props.chartKey}
```

Comment:
you could destructure a lot of vars from `this.props` to reduce the number of `this.props.xxx` calls you make